acceptance/chaos_test.go

```diff
@@ -75,31 +83,36 @@ func TestChaos(t *testing.T) {
 			value := randutil.RandBytes(r, 8192)
 			k := atomic.LoadInt64(&count)
 			var prevErrorOutput string
-			for time.Now().Before(deadline) && atomic.LoadInt32(&stalled) == 0 {
-				clients[i].RLock()
-				v := value[:r.Intn(len(value))]
-				if pErr := clients[i].db.Put(fmt.Sprintf("%08d", k), v); pErr != nil {
-					err := pErr.GoError()
-					if _, ok := err.(*roachpb.SendError); ok || testutils.IsError(err, rpc.ErrShutdown.Error()) || testutils.IsError(err, "client is unhealthy") {
-						// Common errors we can ignore. Also suppress the
-						// log messages so they don't get spammy.
-						curErrorOutput := fmt.Sprintf("client %d: %s", i, err)
-						if prevErrorOutput != curErrorOutput {
-							log.Warning(curErrorOutput)
-							prevErrorOutput = curErrorOutput
+			for !done() {
+				if err := func() error {
```

Comment:
I could but it's going to involve passing in a few parameters like k, value, r, which makes the function awkward.